prometheus/prometheus/handler.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import time
+
+from localstack.aws.api import RequestContext
+from localstack.aws.chain import Handler, HandlerChain
+from localstack.http import Response
+
+from prometheus.metrics.core import (
+    LOCALSTACK_INFLIGHT_REQUESTS_GAUGE,
+    LOCALSTACK_REQUEST_PROCESSING_DURATION_SECONDS,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class TimedRequestContext(RequestContext):
+    start_time: float | None
+
+
+class RequestMetricsHandler(Handler):
+    """
+    Handler that records the start time of incoming requests
+    """
+
+    def __call__(self, chain: HandlerChain, context: TimedRequestContext, response: Response):
+        # Record the start time
+        context.start_time = time.perf_counter()
```

Comment:
Why record here, if we do not try to read it if there is no service_operation set? Shouldn't this be moved after the if?